### PR TITLE
feat: add 'ng promote' command

### PR DESCRIPTION
**Pull request by ashoktamang**
add design doc for 'promote' process

### CODE REVIEW

File: docs/design/promote.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+* The command should only execute inside an Angular project created by angular-cli.
+
+Validation
+---------------
+
+Validation is required for the two arguments in the promote command. It is an important part of the process.
+* Validation executes in three parts which are done in order:
+
+  1. The two arguments of the command (`oldPath` and `newPath`) should always be passed.
+  2. Validtion in `oldPath`
```
Comment (esinek) [inline docs/design/promote.md]: Validtion -> Validation

Comment (ashoktamang) [inline docs/design/promote.md]: Resolved. Thanks.

**Comment by intellix** [comment]
Could this functionality be used for renaming/moving components as well? Sometimes I've created a component and realise I need to rename it. Currently I have to:
 - Rename 4 files: `html`, `css`, `ts`, `spec.ts`
 - Find/replace camelised name in whole project
 - Find/replace dasherized name in whole project

Currently when we generate a component, we do:
```
ng g c somewhere/something
```

It would be nice if you could then move/promote it like: 
```
ng move c somewhere/something somewhere-else/something-else
```
and it would replace all the references and rename everything for you as well. I guess that would also solve promotion right?

**Comment by filipesilva** [comment]
@alxhub what is the status of this PR?

**Comment by hansl** [comment]
Closing this but will keep as a reference.

**Pull request closed without merging**